libs/application/core/src/lib/ApplicationTemplateHelper.ts

```diff
@@ -58,8 +70,22 @@ export class ApplicationTemplateHelper<
    * @param event A state machine event
    * returns [hasChanged, newState, newApplication] where newApplication has the updated state value
    */
-  changeState(event: Event<TEvents>): [boolean, string, Application] {
-    this.initializeStateMachine()
+  changeState(
+    event: Event<TEvents>,
+    emailService: any,
+  ): [boolean, string, Application] {
+    this.initializeStateMachine(undefined, {
+      services: {
+        emailService: async (
+          context: TContext,
+          event: TEvents,
+          { src }: InvokeMeta,
+        ) => {
+          const template = src.template(context)
```

Comment:
maybe use emailTemplate instead as template is used for the application template, just a thought

Reply:
yes, good point